contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,196 @@
+        );
+
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, address(this));
+        _swapForUsdcAndDai(minDaiAmountOut, ratioUSDC, to);
+    }
+
+    /// @notice Function to swap all FEI balance for both DAI and USDC
+    /// @param to the address the DAI and USDC should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+    function swapAllFeiForUsdcAndDai(address to, uint256 ratioUSDC)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 minDaiAmountOut = _redeemAllBalanceFromFeiPSM(address(this));
+        _swapForUsdcAndDai(minDaiAmountOut, ratioUSDC, to);
+    }
+
+    /// @notice Helper function to redeem DAI from the FEI FEI-DAI PSM
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemFromFeiPSM(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address to
+    ) private {
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        feiPSM.redeem(to, amountFeiIn, minDaiAmountOut);
+    }
+
+    /// @notice Helper function to redeem DAI max allowance or balance from the FEI FEI-DAI PSM
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemAllBalanceFromFeiPSM(address to)
+        private
+        returns (uint256 minDaiAmountOut)
+    {
+        uint256 allowance = fei.allowance(msg.sender, address(this));
+        uint256 amountFeiIn = Math.min(fei.balanceOf(msg.sender), allowance);
+
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        uint256 userBalanceBefore = dai.balanceOf(to);
+
+        // 3 is used as there is a 3 basis points redemption fee
+        minDaiAmountOut =
+            amountFeiIn -
+            (amountFeiIn * 3) /
+            Constants.BASIS_POINTS_GRANULARITY;
+        feiPSM.redeem(to, amountFeiIn, minDaiAmountOut);
+
+        uint256 userBalanceAfter = dai.balanceOf(to);
+
+        require(
+            userBalanceAfter - userBalanceBefore >= minDaiAmountOut,
+            "MakerRouter: Not enough DAI received"
+        );
+    }
+
+    /// @notice Helper function to swap for both DAI and USDC
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+
+    function _swapForUsdcAndDai(
+        uint256 minDaiAmountOut,
+        uint256 ratioUSDC,
+        address to
+    ) private {
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        daiPSM.buyGem(to, usdcAmount / USDC_SCALING_FACTOR);
```

Comment:
if usdcAmount / USDC_SCALING_FACTOR == 0, we should revert, otherwise this contract can get stuck with ERC20's

Reply:
if we require that the minimum fei deposited to be 1e18, minDaiAmountOut is most likely always going to be 1e8 or 9.9e17 or something a long those lines, even if ratioUSDC is 1, when we divide that result by the scaling factor, we end up with 90, which is super small, but I think is the smallest output we can get

Reply:
well I guess the above assumption is assuming the minDaiAmountOut is calculated as expected from the FEI PSM, I think we could include the check, wouldn't hurt